packages/aws-cdk/lib/api/hotswap-deployments.ts

```diff
@@ -48,18 +49,58 @@ function findAllHotswappableChanges(
   const hotswappableResources = new Array<HotswapOperation>();
   let foundNonHotswappableChange = false;
   stackChanges.resources.forEachDifference((logicalId: string, change: cfn_diff.ResourceDifference) => {
+    const nonHotswappableResourceFound = isNonHotswappableResourceChange(change);
     const lambdaFunctionShortCircuitChange = isHotswappableLambdaFunctionChange(logicalId, change, assetParamsWithEnv);
-    if (lambdaFunctionShortCircuitChange === ChangeHotswapImpact.REQUIRES_FULL_DEPLOYMENT) {
+    const stepFunctionShortCircuitChange = isHotswappableStepFunctionChange(logicalId, change, assetParamsWithEnv);
+
+    const requiresFullDeployment = (lambdaFunctionShortCircuitChange === ChangeHotswapImpact.REQUIRES_FULL_DEPLOYMENT) ||
+                                   (stepFunctionShortCircuitChange === ChangeHotswapImpact.REQUIRES_FULL_DEPLOYMENT) ||
+                                   (nonHotswappableResourceFound === ChangeHotswapImpact.REQUIRES_FULL_DEPLOYMENT);
+
+    const irrelevant = (lambdaFunctionShortCircuitChange === ChangeHotswapImpact.IRRELEVANT) &&
+                       (stepFunctionShortCircuitChange === ChangeHotswapImpact.IRRELEVANT) &&
+                       (nonHotswappableResourceFound === ChangeHotswapImpact.IRRELEVANT);
+
+    if (requiresFullDeployment) {
       foundNonHotswappableChange = true;
-    } else if (lambdaFunctionShortCircuitChange === ChangeHotswapImpact.IRRELEVANT) {
+    } else if (irrelevant) {
       // empty 'if' just for flow-aware typing to kick in...
     } else {
-      hotswappableResources.push(lambdaFunctionShortCircuitChange);
+      if (typeof lambdaFunctionShortCircuitChange !== 'string') {
+        hotswappableResources.push(lambdaFunctionShortCircuitChange);
+      }
+
+      if (typeof stepFunctionShortCircuitChange !== 'string') {
+        hotswappableResources.push(stepFunctionShortCircuitChange);
+      }
     }
   });
   return foundNonHotswappableChange ? undefined : hotswappableResources;
 }
 
+/**
+ * returns `ChangeHotswapImpact.REQUIRES_FULL_DEPLOYMENT` if a resource was deleted, or a change that we cannot short-circuit occured.
+ * Returns `ChangeHotswapImpact.IRRELEVANT` if a change that does not impact shortcircuiting occured, such as a metadata change.
+ */
+export function isNonHotswappableResourceChange(change: cfn_diff.ResourceDifference): ChangeHotswapImpact {
```

Comment:
Why suddenly the "not" in the name? I think this should be called just `isResourceChangeHotswappabl()`, or something.